platform/api/azure/image.go

```diff
@@ -66,3 +70,53 @@ func IsConflictError(err error) bool {
 	azerr, ok := err.(management.AzureError)
 	return ok && azerr.Code == "ConflictError"
 }
+
+func (a *API) CreateImage(name, resourceGroup, blobURI string) (compute.Image, error) {
+	_, err := a.imgClient.CreateOrUpdate(resourceGroup, name, compute.Image{
+		Name:     &name,
+		Location: &a.opts.Location,
+		ImageProperties: &compute.ImageProperties{
+			StorageProfile: &compute.ImageStorageProfile{
+				OsDisk: &compute.ImageOSDisk{
+					OsType:  compute.Linux,
+					OsState: compute.Generalized,
+					BlobURI: &blobURI,
+				},
+			},
+		},
+	}, nil)
+	if err != nil {
+		return compute.Image{}, err
+	}
+
+	return a.imgClient.Get(resourceGroup, name, "")
+}
+
+// resolveImage is used to ensure that either a Version or DiskURI
+// are provided present for a run. If neither is given via arguments
+// it attempts to parse the Version from the version.txt in the Sku's
+// release bucket.
+func (a *API) resolveImage() error {
+	if a.opts.DiskURI != "" || a.opts.Version != "" {
+		return nil
+	}
+
+	resp, err := http.DefaultClient.Get(fmt.Sprintf("https://%s.release.core-os.net/amd64-usr/current/version.txt", a.opts.Sku))
+	if err != nil {
+		return fmt.Errorf("unable to fetch release bucket %v version: %v", a.opts.Sku, err)
+	}
+
+	scanner := bufio.NewScanner(resp.Body)
```

Comment:
I think I remember seeing parsing this format somewhere else. Should break that out into a util package or something after this PR.